src/context.rs

```diff
@@ -244,6 +244,14 @@ impl FromPeer {
     }
 }
 
+impl Drop for FromPeer {
```

Comment:
This is producing unnecessary diff. Only clone() needed (which too i think it might be no longer as upstream has agreed to not consume it)